File: frontend/src/lib/logger.js
/**
 * Frontend Logger Utility
 */
const Logger = {
  info: (msg, ...args) => {
    if (import.meta.env.DEV) {
      console.log(`[INFO] ${msg}`, ...args);
    }
  },
  warn: (msg, ...args) => {
    console.warn(`[WARN] ${msg}`, ...args);
  },
  error: (msg, ...args) => {
    console.error(`[ERROR] ${msg}`, ...args);
  },
  uiError: (componentName, error, stack) => {
    console.error(`[UI Error in ${componentName}]:`, error, stack);
  }
};

export default Logger;

File: ArduinoTests/Test7_SD/Logger.cpp
#include "Logger.h"
#include "Config.h"

void Logger::info(String msg) {
    if (DEBUG_MODE >= 1) Serial.println("[INFO] " + msg);
}
void Logger::warning(String msg) {
    if (DEBUG_MODE >= 1) Serial.println("[WARN] " + msg);
}
void Logger::error(String msg) {
    if (DEBUG_MODE >= 1) Serial.println("[ERROR] " + msg);
}
void Logger::debug(String msg) {
    if (DEBUG_MODE >= 2) Serial.println("[DEBUG] " + msg);
}
